### PR TITLE
Add missing profiler docs

### PR DESCRIPTION
**Pull request by davmason**
## Summary

Add profiler APIs from 5.0 to the docs

### CODE REVIEW

File: docs/framework/unmanaged-api/profiling/icorprofilerinfo12-eventpipedefineevent-method.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+**Platforms:** See [.NET Core supported operating systems](../../../core/install/windows.md?pivots=os-windows).
+**Header:** CorProf.idl, CorProf.h
+**.NET Versions:** [!INCLUDE[net_core](../../../../includes/net-core-50-md.md)]
+  
+## See also
+
+- [Profiling Interfaces](profiling-interfaces.md)
+- [ICorProfilerCallback10 Interface](icorprofilercallback10-interface.md)
+- [ICorProfilerInfo12 Interface](icorprofilerinfo12-interface.md)
+- [COR_PRF_EVENTPIPE_PARAM_DESC Structure](cor-prf-eventpipe-param-desc-structure.md)
```
Comment (tommcdon) [inline docs/framework/unmanaged-api/profiling/icorprofilerinfo12-eventpipedefineevent-method.md:81]: Suggest adding a link to https://docs.microsoft.com/dotnet/core/diagnostics/eventpipe and https://docs.microsoft.com/dotnet/core/diagnostics/well-known-event-providers

File: docs/framework/unmanaged-api/profiling/icorprofilerinfo12-eventpipecreateprovider-method.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+**Platforms:** See [.NET Core supported operating systems](../../../core/install/windows.md?pivots=os-windows).
+**Header:** CorProf.idl, CorProf.h
+**.NET Versions:** [!INCLUDE[net_core](../../../../includes/net-core-50-md.md)]
+  
+## See also
+
+- [Profiling Interfaces](profiling-interfaces.md)
+- [ICorProfilerCallback10 Interface](icorprofilercallback10-interface.md)
+- [ICorProfilerInfo12 Interface](icorprofilerinfo12-interface.md)
```
Comment (tommcdon) [inline docs/framework/unmanaged-api/profiling/icorprofilerinfo12-eventpipecreateprovider-method.md:43]: Suggest adding a link to https://docs.microsoft.com/dotnet/core/diagnostics/eventpipe and https://docs.microsoft.com/dotnet/core/diagnostics/well-known-event-providers

File: docs/framework/unmanaged-api/profiling/icorprofilerinfo12-eventpipeaddprovidertosession-method.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+**Platforms:** See [.NET Core supported operating systems](../../../core/install/windows.md?pivots=os-windows).
+**Header:** CorProf.idl, CorProf.h
+**.NET Versions:** [!INCLUDE[net_core](../../../../includes/net-core-50-md.md)]
+  
+## See also
+
+- [Profiling Interfaces](profiling-interfaces.md)
+- [ICorProfilerCallback10 Interface](icorprofilercallback10-interface.md)
+- [ICorProfilerInfo12 Interface](icorprofilerinfo12-interface.md)
+- [COR_PRF_EVENTPIPE_PROVIDER_CONFIG Structure](cor-prf-eventpipe-provider-config-structure.md)
```
Comment (tommcdon) [inline docs/framework/unmanaged-api/profiling/icorprofilerinfo12-eventpipeaddprovidertosession-method.md:44]: Suggest adding a link to https://docs.microsoft.com/dotnet/core/diagnostics/eventpipe

File: docs/framework/unmanaged-api/profiling/icorprofilerinfo12-eventpipegetproviderinfo-method.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+**Platforms:** See [.NET Core supported operating systems](../../../core/install/windows.md?pivots=os-windows).
+**Header:** CorProf.idl, CorProf.h
+**.NET Versions:** [!INCLUDE[net_core](../../../../includes/net-core-50-md.md)]
+  
+## See also
+
+- [Profiling Interfaces](profiling-interfaces.md)
+- [ICorProfilerCallback10 Interface](icorprofilercallback10-interface.md)
+- [ICorProfilerInfo12 Interface](icorprofilerinfo12-interface.md)
```
Comment (tommcdon) [inline docs/framework/unmanaged-api/profiling/icorprofilerinfo12-eventpipegetproviderinfo-method.md:52]: Suggest adding a link to https://docs.microsoft.com/dotnet/core/diagnostics/eventpipe and https://docs.microsoft.com/dotnet/core/diagnostics/well-known-event-providers

File: docs/framework/unmanaged-api/profiling/icorprofilerinfo12-eventpipestartsession-method.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+**Platforms:** See [.NET Core supported operating systems](../../../core/install/windows.md?pivots=os-windows).
+**Header:** CorProf.idl, CorProf.h
+**.NET Versions:** [!INCLUDE[net_core](../../../../includes/net-core-50-md.md)]
+  
+## See also
+
+- [Profiling Interfaces](profiling-interfaces.md)
+- [ICorProfilerCallback10 Interface](icorprofilercallback10-interface.md)
+- [ICorProfilerInfo12 Interface](icorprofilerinfo12-interface.md)
```
Comment (tommcdon) [inline docs/framework/unmanaged-api/profiling/icorprofilerinfo12-eventpipestartsession-method.md:52]: same as above

File: docs/framework/unmanaged-api/profiling/icorprofilerinfo12-interface.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+## Requirements  
+
+**Platforms:** See [.NET Core supported operating systems](../../../core/install/windows.md?pivots=os-windows).  
+**Header:** CorProf.idl, CorProf.h  
+**.NET Versions:** [!INCLUDE[net_core](../../../../includes/net-core-50-md.md)]  
+
+## See also
+
+- [Profiling Interfaces](profiling-interfaces.md)
+- [ICorProfilerCallback10 Interface](icorprofilercallback10-interface.md)
```
Comment (tommcdon) [inline docs/framework/unmanaged-api/profiling/icorprofilerinfo12-interface.md:38]: Suggest adding a link to https://docs.microsoft.com/dotnet/core/diagnostics/eventpipe and https://docs.microsoft.com/dotnet/core/diagnostics/well-known-event-providers

**Review by tommcdon — APPROVED** [review]
Left a few comments, otherwise LGTM

File: includes/net-core-50-md.md [hunk 7]
```diff
@@ -0,0 +1 @@
+Available since .NET Core 5.0
```
Comment (sywhang) [inline includes/net-core-50-md.md:1]:
```suggestion
Available since .NET 5.0
```

**Review by sywhang — APPROVED** [review]
Thanks @davmason! I can merge this after you address the feedback here :-)

**Comment by davmason** [comment]
Don't merge for a bit, I just discovered the we use wildy different formatting in some areas of the profiler docs and I want to fix it

**Comment by sywhang** [comment]
> Don't merge for a bit, I just discovered the we use wildy different formatting in some areas of the profiler docs and I want to fix it

Sure, just let me know when it's ready for review then

**Comment by davmason** [comment]
@sywhang can you take a look now? We had 3-4 different ways we were formatting parameters, I picked what seemed to be the most common one and made them all consistent

**Review by sywhang — APPROVED** [review]
:shipit:

**Pull request merged**